internal/infrastructure/mongo/mongodoc/client.go

```diff
@@ -312,6 +290,57 @@ func (c *Client) Paginate(ctx context.Context, col string, filter interface{}, p
 	return usecase.NewPageInfo(int(count), startCursor, endCursor, hasNextPage, hasPreviousPage), nil
 }
 
+func sortOptionsFrom(sort *string, p *Pagination, key string) (bson.D, string) {
+	var sortOptions bson.D
+	var sortKey = ""
+	if sort != nil && len(*sort) > 0 && *sort != "id" {
+		sortKey = *sort
+		sortOptions = append(sortOptions, bson.E{Key: sortKey, Value: p.SortDirection()})
+	}
+	sortOptions = append(sortOptions, bson.E{Key: key, Value: p.SortDirection()})
+	return sortOptions, sortKey
+}
+
+func paginationFilter(ctx context.Context, coll *mongo.Collection, p *Pagination, sortKey, key string, filter bson.M) (bson.M, *int64, error) {
```

Comment:
When a function returns a reference, it means that the responsibility for its nil check is delegated to the caller of the function. Forgetting this can lead to panic. In this case, it is not necessary that the limit be a reference, since the limit will simply be ignored if an err is encountered.

```suggestion
func paginationFilter(ctx context.Context, coll *mongo.Collection, p *Pagination, sortKey, key string, filter bson.M) (bson.M, int64, error) {
```